### DISCUSSION TITLE
Is there a way to merge the enter and open in keymap?

### DISCUSSION THREAD

**Discussion by OriLiMu** [sxyazi/yazi#451] [1. Q&A] [answered]
As title said. For example, I have a keymap 'l', when press l on folder, will enter it, when press on file will open it.

**Comment by sxyazi** [comment, marked as answer]
See https://yazi-rs.github.io/docs/faq#why-cant-open-and-enter-be-a-single-command

When the plugin system is available (should be soon, https://github.com/sxyazi/yazi/pull/401), for users who *truly* don't need it, it will be easy to implement with just a few lines of code.

**Reply by evanescente-ondine** [reply]
Is it functional yet ? I added the lines but it doesn't work.
I understand the reasoning behind that choice but some switch to be hardcoded.
Or better, do something broot does:
> 	{
         key: enter
	 internal: "focus {directory}"
	 apply_to: directory
	}

That would be an incredibly simple and practical feature. Then if it breaks the other shortcuts... let the user deal with it. and you put different extensions and tailor the command to them. Sort of a generalized opener that also executes internal commands;

**Reply by sxyazi** [reply]
Yeah see https://yazi-rs.github.io/docs/tips#smart-enter-enter-for-directory-open-for-file

**Comment by evanescente-ondine** [comment]
Sorry I just ried to remove that comment but couldn't find it.
My repo version wasn't up to date. Thanks ;-)
